eva/interfaces/relational/relation.py

```diff
@@ -0,0 +1,220 @@
+import pandas
+from eva.interfaces.relational.utils import (
+    create_limit_expression,
+    create_star_expression,
+    execute_statement,
+    handle_select_clause,
+    sql_predicate_to_expresssion_tree,
+    sql_string_to_expresssion_list,
+    string_to_lateral_join,
+    try_binding,
+)
+from eva.models.storage.batch import Batch
+from eva.parser.alias import Alias
+from eva.parser.select_statement import SelectStatement
+
+from eva.parser.statement import AbstractStatement
+from eva.parser.table_ref import JoinNode, TableRef
+from eva.parser.types import JoinType
+from eva.parser.utils import parse_sql_orderby_expr
+
+
+class EVARelation:
+    def __init__(
+        self, query_node: Union[AbstractStatement, TableRef], alias: Alias = None
+    ):
+        self._query_node = query_node
+        self._alias = alias
+        self._dummy_alias = Alias("relation")
+
+    def alias(self, alias: str) -> "EVARelation":
+        """Returns a new Relation with an alias set.
+
+        Args:
+            alias (str): an alias name to be set for the Relation.
+
+        Returns:
+            EVARelation: Aliased Relation.
+
+        Examples:
+            >>> relation = conn.table("sample_table")
+            >>> relation.alias('table')
+        """
+        self._alias = Alias(alias)
+
+    def cross_apply(self, expr: str, alias: str) -> "EVARelation":
+        """Execute a expr on all the rows of the relation
+
+        Args:
+            expr (str): sql expression
+            alias (str): alias of the output of the expr
+
+        Returns:
+            `EVARelation`: relation
+
+        Examples:
+
+            Runs Yolo on all the frames of the input table
+
+            >>> relation = conn.table("videos")
+            >>> relation.cross_apply("Yolo(data)", "objs(labels, bboxes, scores)")
+
+            Runs Yolo on all the frames of the input table and unnest each object as separate row.
+
+            >>> relation.cross_apply("unnest(Yolo(data))", "obj(label, bbox, score)")
+        """
+        assert self._query_node.from_table is not None
+
+        table_ref = string_to_lateral_join(expr, alias=alias)
+        join_table = TableRef(
+            JoinNode(
+                TableRef(self._query_node, alias=self._alias),
+                table_ref,
+                join_type=JoinType.LATERAL_JOIN,
+            )
+        )
+        self._query_node = SelectStatement(
+            target_list=create_star_expression(), from_table=join_table
+        )
+        # reset the alias as after join there isn't a single alias
+        self._alias = Alias("Relation")
+        try_binding(self._query_node)
+        return self
+
+    def df(self) -> pandas.DataFrame:
+        """Execute and fetch all rows as a pandas DataFrame
+
+        Returns:
+            pandas.DataFrame:
+        """
+        batch = self.execute()
+        assert batch is not None, "relation execute failed"
+        return batch.frames
+
+    def execute(self) -> Batch:
```

Comment:
I understand this function is needed for trigger execution, but I am not sure whether return a `Batch` to users is useful.

Instead, I think providing a `show` function like other systems is more straightforward, which just directly prints results.

Reply:
We have `.df()`. I earlier had `show` as well

Reply:
Added `show()`